docs/source/cpp/compute.rst

```diff
@@ -575,6 +575,22 @@ when a positive ``max_splits`` is given.
   as separator.
 
 
+String joining
+~~~~~~~~~~~~~~
+
+The inverse of string splitting:
+
++--------------------------+------------+-------------------------+-------------------+---------+
+| Function name            | Arity      | Input types             | Output type       | Notes   |
++==========================+============+=========================+===================+=========+
+| binary_join              | Binary     | List-like               | String-like       | \(1)    |
++--------------------------+------------+-------------------------+-------------------+---------+
+
+* \(1) First argument must be an array, while the second argument (`separator`)
+can be a scalar or array. When the `separator` argument is an array, the join is
+performed by each corresponding element.
```

Comment:
Be careful with indentation in bullet lists, you need the continuation lines to be indented as the first line after the `*`:
```
* \(1) First argument must be an array, while the second argument (`separator`)
  can be a scalar or array. When the `separator` argument is an array, the join is
  performed by each corresponding element.
```